treeherder/model/management/commands/cycle_data.py

```diff
@@ -132,15 +133,12 @@ def cycle(self):
     def _remove_leftovers(self):
         # remove any signatures which are
         # no longer associated with a job
-        logger.warning('Removing performance signatures with missing jobs...')
-        for signature in PerformanceSignature.objects.filter(last_updated__lte=self.max_timestamp):
-            self._quit_on_timeout()
-
-            if not PerformanceDatum.objects.filter(
-                repository_id=signature.repository_id,  # leverages (repository, signature) compound index
-                signature_id=signature.id,
-            ).exists():
-                signature.delete()
+        signatures = PerformanceSignature.objects.filter(last_updated__lte=self.max_timestamp)
+        tc_model = TaskclusterModel(
+            root_url, client_id=settings.NOTIFY_CLIENT_ID, access_token=settings.NOTIFY_ACCESS_TOKEN
```

Comment:
I'm curious why we're using the taskcluster notify service to send an email about deleted signatures. Could this be done with this python package (or some other package)? https://docs.python.org/3/library/smtplib.html

Reply:
Me & @camd synced about what to use for emailing. As Treeherder doesn't do any emailing, we concluded the Taskcluster team might provide a hint, as they already have a service at hand
I contacted Peter Moore which confirmed that & further guided me into using Taskcluster's standalone Notify service for emailing.

I was aware of Python builtin smtplib, however I wasn't sure how that would have played with Gmail's spam filtering. So I avoided this approach.

Reply:
Ok, thanks for the explanation. I've been talking with @beatrice-acasandrei about the env variables/credentials for this notify service and we'll need to see if we can get treeherder's deployments registered instead of using her personal credentials. So that will be a blocker to getting this merged. I'll reach out to Peter Moore/taskcluster team about this and loop in you and Beatrice.

Reply:
Super, thanks for helping us with the proper credentials!